base/util.jl

```diff
@@ -221,6 +221,7 @@ function julia_cmd(julia=joinpath(Sys.BINDIR, julia_exename()))
         push!(addflags, "--sysimage-native-code=no")
     end
     if opts.use_pkgimage_native_code == 0
+        @assert opts.malloc_log == 0 && opts.code_coverage == 0
```

Comment:
@vchuravy This assert is causing [this test](https://github.com/JuliaLang/Pkg.jl/blob/ed6a5497e46ed541b2718c404c0f468b7f92263a/test/pkg.jl#L236) to fail.  While [the error from `Base.runtests()` as seen on CI](https://buildkite.com/julialang/julia-master/builds/17904#018443d1-9bcb-4557-9697-27de29a9400a/669-1179) is not so clear, if you run this MWE:

```
using Pkg, Test
using Base: UUID

include(joinpath(Sys.STDLIB, "Pkg", "test", "utils.jl"))
using .Utils

const TEST_PKG = (name = "Example", uuid = UUID("7876af07-990d-54b4-ab0e-23690620f79a"))
temp_pkg_dir() do project_path
    Pkg.add(TEST_PKG.name)
    Pkg.test(TEST_PKG.name; coverage=true)
end
```

You get the much clearer error:
```
ERROR: LoadError: AssertionError: opts.malloc_log == 0 && opts.code_coverage == 0
Stacktrace:
  [1] julia_cmd(julia::String)
    @ Base ./util.jl:224
  [2] julia_cmd
    @ ./util.jl:153 [inlined]
  [3] create_expr_cache(pkg::Base.PkgId, input::String, output::String, output_o::Nothing, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, internal_stderr::IO, internal_stdout::IO)
    @ Base ./loading.jl:1735
  [4] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, keep_loaded_modules::Bool)
    @ Base ./loading.jl:1841
  [5] compilecache
    @ ./loading.jl:1807 [inlined]
  [6] _require(pkg::Base.PkgId, env::String)
    @ Base ./loading.jl:1452
  [7] _require_prelocked(uuidkey::Base.PkgId, env::String)
    @ Base ./loading.jl:1316
  [8] macro expansion
    @ ./loading.jl:1296 [inlined]
  [9] macro expansion
    @ ./lock.jl:267 [inlined]
 [10] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1259
 [11] include(fname::String)
    @ Base.MainInclude ./client.jl:478
 [12] top-level scope
    @ none:6
in expression starting at /tmp/jl_nzdL87/packages/Example/aqsx3/test/runtests.jl:1
ERROR: LoadError: Package Example errored during testing
Stacktrace:
  [1] pkgerror(msg::String)
    @ Pkg.Types /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/Types.jl:68
  [2] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, julia_args::Cmd, test_args::Cmd, test_fn::Nothing, force_latest_compatible_version::Bool, allow_earlier_backwards_compatible_versions::Bool, allow_reresolve::Bool)
    @ Pkg.Operations /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/Operations.jl:1878
  [3] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, test_fn::Nothing, julia_args::Cmd, test_args::Cmd, force_latest_compatible_version::Bool, allow_earlier_backwards_compatible_versions::Bool, allow_reresolve::Bool, kwargs::Base.Pairs{Symbol, Base.TTY, Tuple{Symbol}, NamedTuple{(:io,), Tuple{Base.TTY}}})
    @ Pkg.API /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/API.jl:435
  [4] kwcall(::NamedTuple{(:coverage, :io), Tuple{Bool, Base.TTY}}, ::typeof(Pkg.API.test), ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/API.jl:414
  [5] test(pkgs::Vector{Pkg.Types.PackageSpec}; io::Base.TTY, kwargs::Base.Pairs{Symbol, Bool, Tuple{Symbol}, NamedTuple{(:coverage,), Tuple{Bool}}})
    @ Pkg.API /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/API.jl:156
  [6] kwcall(::NamedTuple{(:coverage,), Tuple{Bool}}, ::typeof(Pkg.API.test), pkgs::Vector{Pkg.Types.PackageSpec})
    @ Pkg.API /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/API.jl:145
  [7] #test#75
    @ /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/API.jl:144 [inlined]
  [8] #test#74
    @ /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/src/API.jl:143 [inlined]
  [9] (::var"#3#4")(project_path::String)
    @ Main /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/test.jl:10
 [10] (::Main.Utils.var"#14#15"{Bool, var"#3#4"})()
    @ Main.Utils /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/test/utils.jl:148
 [11] withenv(::Main.Utils.var"#14#15"{Bool, var"#3#4"}, ::Pair{String, Nothing}, ::Vararg{Pair{String, Nothing}})
    @ Base ./env.jl:197
 [12] temp_pkg_dir(fn::var"#3#4"; rm::Bool, linked_reg::Bool)
    @ Main.Utils /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/test/utils.jl:140
 [13] temp_pkg_dir(fn::Function)
    @ Main.Utils /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/share/julia/stdlib/v1.9/Pkg/test/utils.jl:119
 [14] top-level scope
    @ /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/test.jl:8
in expression starting at /build/artifacts/julia-f9063bd99e-linux-x86_64/julia-f9063bd99e/test.jl:8
```

Reply:
Yeah the assert is wrong. It should have been pkgimage = 1 => malloc_log= false

Reply:
This now fails with:

```
ERROR: LoadError: AssertionError: ocachepath === nothing                                                                                                                                                                    
Stacktrace:                                                                                                   
 [1] _tryrequire_from_serialized(pkg::Base.PkgId, path::String, ocachepath::String)                                                                                                                                         
   @ Base ./loading.jl:1073                                                                                   
 [2] _require(pkg::Base.PkgId, env::String)
   @ Base ./loading.jl:1463                                                                                   
 [3] _require_prelocked(uuidkey::Base.PkgId, env::String)
   @ Base ./loading.jl:1316                                                                                   
 [4] macro expansion                     
   @ ./loading.jl:1296 [inlined] 
 [5] macro expansion                               
   @ ./lock.jl:267 [inlined]                                                                                  
 [6] require(into::Module, mod::Symbol)                                                                                                                                                                                     
   @ Base ./loading.jl:1259
 [7] include(fname::String)                                                                                   
   @ Base.MainInclude ./client.jl:478                                                                         
 [8] top-level scope            
   @ none:6                                                                                                   
in expression starting at /tmp/jl_0wBWiO/packages/Example/aqsx3/test/runtests.jl:1
```